File: playground/kwj1270/commerce/display/src/main/java/team/sipe/commerce/display/DisplayApi.java
package team.sipe.commerce.display;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import team.sipe.commerce.display.application.DeliveryClient;
import team.sipe.commerce.display.application.OrderClient;

import java.util.List;

@RequestMapping("/display")
@RestController
public class DisplayApi {

    private final OrderClient orderClient;
    private final DeliveryClient deliveryClient;

    public DisplayApi(final OrderClient orderClient, final DeliveryClient deliveryClient) {
        this.orderClient = orderClient;
        this.deliveryClient = deliveryClient;
    }

    @GetMapping("/{orderId}")
    public ResponseEntity<DisplayOrderInfoHttpResponse> displayOrderInfo(@PathVariable final String orderId) {
        return ResponseEntity.ok(new DisplayOrderInfoHttpResponse(List.of()));
    }

    record DisplayOrderInfoHttpResponse(
            List<DisplayOrderItem> orderItems
    ) {
    }

    record DisplayOrderItem(
            DisplayProductBundleDetail productBundleDetail,
            DisplayProductDetail productDetail,
            DisplayDeliveryDetail deliveryDetail,
            DisplayRefundDetail refundDetail
    ) {
    }

    record DisplayProductBundleDetail() {
    }

    record DisplayProductDetail() {
    }

    record DisplayDeliveryDetail() {
    }

    record DisplayRefundDetail() {
    }
}

File: playground/kwj1270/commerce/display/src/main/kotlin/team/sipe/display/api/DisplayApi.kt
package team.sipe.display.api

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import team.sipe.display.core.DisplayQueryService
import team.sipe.display.core.dto.DisplayQueryInformation

@RequestMapping("/display")
@RestController
class DisplayApi(
    private val displayQueryService: DisplayQueryService
) {

    @GetMapping("/orders/{orderId}")
    fun orders(@PathVariable("orderId") orderId: Long): ResponseEntity<DisplayQueryInformation> {
        val response = displayQueryService.findById(orderId)
        return ResponseEntity.ok(response)
    }
}
